addons/mail/static/src/model/model_manager.js

```diff
@@ -35,18 +35,20 @@ class ModelManager {
          */
         this.env = env;
         /**
-         * Whether this is currently handling an "update after" on a record.
-         * Useful to determine if we should process computed/related fields.
+         * Set of records that have been created during the current update
+         * cycle. Useful to trigger _created.
          */
-        this._isHandlingToUpdateAfters = false;
+        this._createdRecords = new Set();
         /**
-         * Determine whether an update cycle is currently in progress.
-         * Useful to determine whether an update should initiate an update
-         * cycle or not. An update cycle basically prioritizes processing
-         * of all direct updates (i.e. explicit from `data`) before
-         * processing computes.
+         * Whether any change happened during the current update cycle. Useful
+         * to know if rev number has to be incremented.
          */
-        this._isInUpdateCycle = false;
+        this._hasAnyChangeDuringCycle = false;
+        /**
+         * Set of records that have been updated during the current update
+         * cycle. Useful to increment record rev numbers.
+         */
+        this._updatedRecords = new Set();
```

Comment:
Suggestion to prefix attributes related to dynamic aspect of update cycle with `_uc`:
```
_ucUpdateAfters
_ucAreChanges
_ucComputes
_ucCreatedRecords
_ucUpdatedRecords
```

This way, they are defined together (nested object makes things too long and annoying to use, hence preference of flat attributes), shorter name for `_hasAnyChangeDuringCycle`, and I think it's easier to get `_uc` variables are dynamic and "live" in the context of a single update cycle.

Reply:
I'm surprised that you suggest an abbreviation. Shouldn't we just have an _updateCycle object with various keys?